libcxx/test/std/iterators/iterator.requirements/indirectcallable/indirectinvocable/indirect_binary_predicate.compile.pass.cpp

```diff
@@ -80,3 +80,11 @@ struct BadPredicate6 {
     bool operator()(std::iter_common_reference_t<It1>, std::iter_common_reference_t<It2>) const = delete;
 };
 static_assert(!std::indirect_binary_predicate<BadPredicate6, It1, It2>);
+
+// Test ADL-proofing (P2538R1)
+#if TEST_STD_VER >= 26
```

Comment:
Why are we not testing this in older dialectts?

Reply:
These tests pass before your change as well.

Also you're missing an include for `test_macros.h`, so this test was never running.

Reply:
Good catch with the `test_macros.h`.

> Why are we not testing this in older dialectts?

This is technically a C++26 paper, so I was going to test this only in C++26. The fact that it happens to pass in C++20/23 is just because we will implement `projected` the same in all standard modes, but we're not technically guaranteeing that `projected` is ADL-proof in older standards. Does that make sense?

Reply:
Regarding `test_macros.h`, `-Wundef` actually catches those!

Reply:
This very much feels like a defect report. I would rather we test the behavior in all dialects where we provide the behavior.